packages/path_provider/path_provider_windows/example/integration_test/path_provider_test.dart

```diff
@@ -14,32 +12,36 @@ void main() {
 
   testWidgets('getTemporaryDirectory', (WidgetTester tester) async {
     final PathProviderWindows provider = PathProviderWindows();
-    final String result = await provider.getTemporaryPath();
+    final String? result = await provider.getTemporaryPath();
     _verifySampleFile(result, 'temporaryDirectory');
   });
 
   testWidgets('getApplicationDocumentsDirectory', (WidgetTester tester) async {
     final PathProviderWindows provider = PathProviderWindows();
-    final String result = await provider.getApplicationDocumentsPath();
+    final String? result = await provider.getApplicationDocumentsPath();
     _verifySampleFile(result, 'applicationDocuments');
   });
 
   testWidgets('getApplicationSupportDirectory', (WidgetTester tester) async {
     final PathProviderWindows provider = PathProviderWindows();
-    final String result = await provider.getApplicationSupportPath();
+    final String? result = await provider.getApplicationSupportPath();
     _verifySampleFile(result, 'applicationSupport');
   });
 
   testWidgets('getDownloadsDirectory', (WidgetTester tester) async {
     final PathProviderWindows provider = PathProviderWindows();
-    final String result = await provider.getDownloadsPath();
+    final String? result = await provider.getDownloadsPath();
     _verifySampleFile(result, 'downloads');
   });
 }
 
 /// Verify a file called [name] in [directoryPath] by recreating it with test
 /// contents when necessary.
-void _verifySampleFile(String directoryPath, String name) {
+void _verifySampleFile(String? directoryPath, String name) {
+  expect(directoryPath, isNotNull);
+  if (directoryPath == null) {
+    return;
```

Comment:
nit: is this needed?

Reply:
'expect' doesn't cause type promotion, so it's either this or a `!`. I thought this would be clearer in case of failure than just doing `!`.

Reply:
ah - I see.  One possible alternative is to move the `expect` before the call and then have this parameter be non-null.

Reply:
That just takes the same problem and repeats it more times though.